packages/director/src/lib/hooksReporter.ts

```diff
@@ -0,0 +1,207 @@
+import { hookEvents, hookTypes } from '@src/lib/hooksEnums';
+import axios from 'axios';
+import { getDashboardRunURL } from '@src/lib/urls';
+import { Instance } from '@src/types/instance.types';
+import { Run } from '@src/types/run.types';
+import { Project, Hook } from '@src/types/project.types';
+import { hookReportSchema } from '@src/lib/schemas';
+import Ajv from 'ajv';
+
+const ajv = new Ajv({ removeAdditional: 'all' });
+const cleanHookReportData = ajv.compile(hookReportSchema);
```

Comment:
Interesting, never used before 👀

Reply:
This is kinda a weird case. Typescript is kinda succy when you wanna create an object on the fly inside a function. So I used a schema validator to remove the un wanted/needed fields from an object with an already defined type. The main reason this was needed was the data set being sent back to the webhooks was too large. Node was flipping out about the size.

Honestly I think using plain javascript to build the object on the fly is cleaner but again typescript makes that dang near impossible.